test/geoloco/Geoloco_Regression.ipynb

```diff
@@ -0,0 +1,343 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ad598301-433b-41ae-850e-3caa8968a873",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if (mod021km_geoloco_info is not None and \n",
+    "    mod35l2_geoloco_info is not None and\n",
+    "    mod08d3_geoloco_info is not None):\n",
+    "\n",
+    "    mod021km_request = Request(collection=mod021km_geoloco_info['collection'],\n",
+    "                               granule_id=mod021km_geoloco_info['granule_id'],\n",
+    "                               variables=mod021km_geoloco_info['variable'],\n",
+    "                               scale_size=mod021km_geoloco_info['downscale_size'],\n",
+    "                               crs=geo_proj4_string,\n",
+    "                               interpolation=resampling_string)\n",
+    "\n",
+    "    mod35l2_request = Request(collection=mod35l2_geoloco_info['collection'],\n",
+    "                              granule_id=mod35l2_geoloco_info['granule_id'],\n",
+    "                              variables=mod35l2_geoloco_info['variable'],\n",
+    "                              scale_size=mod35l2_geoloco_info['downscale_size'],\n",
+    "                              crs=geo_proj4_string,\n",
+    "                              interpolation=resampling_string)\n",
+    "\n",
+    "    mod08d3_request = Request(collection=mod08d3_geoloco_info['collection'],\n",
+    "                              granule_id=mod08d3_geoloco_info['granule_id'],\n",
+    "                              variables=mod08d3_geoloco_info['variable'],\n",
+    "                              scale_size=mod08d3_geoloco_info['downscale_size'],\n",
+    "                              crs=geo_proj4_string,\n",
+    "                              interpolation=resampling_string)\n",
+    "\n",
+    "    mod021km_compare_file = submit_and_download(harmony_client, mod021km_request, file_indicators['MOD021KM'])\n",
+    "    mod35l2_compare_file = submit_and_download(harmony_client, mod35l2_request, file_indicators['MOD35_L2'])\n",
+    "    mod08d3_compare_file = submit_and_download(harmony_client, mod08d3_request, file_indicators['MOD08_D3'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8cb23cdb-3211-49c3-8d98-7351feb35c44",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if (mod021km_geoloco_info is not None and \n",
+    "    mod35l2_geoloco_info is not None and\n",
+    "    mod08d3_geoloco_info is not None):\n",
+    "    \n",
+    "    geoloco_tests = 1\n",
+    "    \n",
+    "    mod021km_reference_dims = get_dim_sizes(reference_data['MOD021KM'])\n",
+    "    mod021km_dims = get_dim_sizes(mod021km_compare_file)\n",
+    "    if mod021km_reference_dims != mod021km_dims:\n",
+    "        \"\"\"if mod021km_reference_dims == mod021km_dims:\n",
```

Comment:
Okay - so bit of a rollercoaster ahead...

First up - it looks like things are much quicker now you consolidated down to fewer tests and they all run within about 2:30 on SIT and UAT. Nice!

Next up - I tested what happened when I set `geoloco_tests = 1` in the first comparison block. Because the subsequent tests reset the value of `geoloco_tests = 0`, this test failure wasn't picked up, so the notebook still ended saying everything passed. Maybe consider keeping separate variables for each test (`mod021km_test_pass`, `mod35l2_test_pass` and `mod08d3_test_pass` maybe) and then having a check at the end that all three are equal to the value you expect. (Also, a personal preference here would be to use a boolean value of `True` or `False`, rather than `0` and `1` - it's just a tiny bit more readable)

I like having the code ready in place to add the ability to compare the output arrays when you are ready. But a couple of thoughts here:

1) Channelling Matt a bit - it's probably better to use regular comments (with `#`) rather than a multi-line string, to do so. (For what it's worth, this is something I totally used to do, but I think Matt's right that it's better to not)
2) I'd consider making a new utility function that takes the path to the reference file and the downloaded output file, and then within that function does the comparison. It'll make the notebook a bit tidier, and make maintenance of that function a lot easier (making one set of code changes if that function needs to be updated, but improving all the places it is called)
3) I think you could use something like [numpy.array_equal](https://numpy.org/doc/stable/reference/generated/numpy.array_equal.html) or [numpy.testing.assert_array_equal](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html) to simplify the array comparison.
4) Bit of a nit - maybe having an in-line comment near the code that is commented out to make it clear why it's commented out, and that it is ready to be used in the near future?

Sorry - bit of an essay there!